test/metabase/models/on_demand_test.clj

```diff
@@ -14,7 +14,7 @@
    [toucan2.core :as t2]
    [toucan2.tools.with-temp :as t2.with-temp]))
 
-(defn- do-with-mocked-field-values-updating
+(defn- do-with-mocked-field-values-updating!
```

Comment:
I split up some mega tests into a series of smaller ones and also added `!` to some of the destructive functions here.